sdk/src/wallet/core/operations/client.rs

```diff
@@ -61,6 +60,16 @@ where
         {
             *self.client.mqtt.broker_options.write().await = broker_options;
         }
+
+        // Update the protocol of the network_info to not have the default data, which can be wrong
```

Comment:
I don't understand getting the info. Isn't this kind of up to to caller?

Reply:
Usually people will just build the client options like this `ClientOptions::new().with_node("https://api.testnet.shimmer.network")?`, so basically only set the node field and everything else will be our default values. But the default values for protocol params are wrong if it's not the shimmer network, therefore the addresses will also have `smr` in the accounts, even though the network for which you provided the url, can have something else

Reply:
then we should change the protocol params field to be optional so we know whether it's been set manually by the user

Reply:
But this would be breaking change

Reply:
Well, technically yes, but I wonder if it's necessary to fix the bug here. This way isn't correct, because it overwrites the user's manually defined options.

Reply:
We overwrite this already anyways in the node syncing process. The background syncing just runs every x seconds and therefore highly unlikely at exactly this moment where we need to have it updated